packages/instantsearch.js/src/lib/InstantSearch.ts

```diff
@@ -565,16 +576,23 @@ See documentation: ${createDocumentationLink({
     // we need to respect this helper as a way to keep all listeners correct.
     const mainHelper =
       this.mainHelper ||
-      algoliasearchHelper(this.client, this.indexName, undefined, {
-        persistHierarchicalRootCount: this.future.persistHierarchicalRootCount,
-      });
+      algoliasearchHelper(
+        this.client,
+        this.indexName,
+        this.compositionID ? { compositionID: this.compositionID } : undefined,
```

Comment:
this should be in the index derived helper instead of here, I'm not convinced these parameters survive `refreshUiState` for example

Reply:
as discussed since then, in order to be out-of-the-box compatible with the sort-by widget, we had to walk away from the option of exposing `compositionID` in the Search Parameters in favor of relying solely on the existing `index` query parameter to store the compositionID